---
title: "ConVarT_orthoFinder"
output: markdown_github
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r, echo=FALSE, include=FALSE}
library(data.table)
msa<-fread("data/example_msa_input.txt")
output<-fread("data/example_output.txt")
input1<-fread("data/example_df1_input.txt")
input2<-fread("data/example_df2_input.txt")
```

`orthoFinder` function can be used to find orthologous variants between species. `msa` file generated by `ConVarT_Pipeline` is used as an input. Other required files are:

-   Variant data as `data.table` or `data.frame`, with at least following columns:

    -   `Refseq_ID`: NCBI reference sequence id of proteins.

    -   `aapos`: Amino acid position of variants.

    -   `from`: Reference amino acid.

    -   `to`: Converting amino acid.

```{r, echo=FALSE, results='asis'}
library(knitr)
kable(input2[1:5,])
```

-   `msa` table needs to be modified according to `msa_modification.R` script. Below are example modified data.

```{r, echo=FALSE, results='asis'}
kable(msa[1:5,])
```

## `orthoFinder` function

The function needs following inputs:

-   `df1`: Variant data for the first organism

-   `df2`: Variant data for the second organism

-   `org1`: Name of the first organism. This needs to be the same with the column name of modified `msa` data

-   `org2`: Same as `org1`, for second organism.

-   `msa`: Modified msa table

-   `ort`: Should data be filtered according to type of variant (conserved and non-conserved). Default is `TRUE`. Refer to paper for detailed explanation: [<https://doi.org/10.1101/2021.01.07.424951>](https://doi.org/10.1101/2021.01.07.424951){.uri}

Output is `data.table` with each row representing a variant-orthologous variant combination. Below is an example output:

```{r, echo=FALSE, results='asis'}
library(knitr)
kable(output[1:5,])
```
